src/wasm/wasm-debug.cpp

```diff
@@ -99,26 +103,318 @@ void dumpDWARF(const Module& wasm) {
 //     StringMap<std::unique_ptr<MemoryBuffer>>
 //     EmitDebugSections(llvm::DWARFYAML::Data &DI, bool ApplyFixups);
 //
-// For modifying data, like line numberes, we can in theory do that either on
-// the DWARFContext or DWARFYAML::Data; unclear which is best, but modifying
-// the DWARFContext may save us doing fixups in EmitDebugSections.
-//
 
-void writeDWARFSections(Module& wasm) {
+// Represents the state when parsing a line table.
```

Comment:
where in LLVM is it that does this same logic?

Reply:
I think this is the main place: https://github.com/WebAssembly/binaryen/blob/bd4cac987f19ee4f59b161a77b996ff1de46c4b9/third_party/llvm-project/DWARFDebugLine.cpp#L531